cassandra/src/main/scala/org/apache/spark/sql/cassandra/CassandraXDSourceRelation.scala

```diff
@@ -132,32 +136,69 @@ class CassandraXDSourceRelation(
   }
 
   override def sizeInBytes: Long = {
-    //TODO  Retrieve table size from C* system table from Cassandra 2.1.4
     // If it's not found, use SQLConf default setting
     tableSizeInBytes.getOrElse(sqlContext.conf.defaultSizeInBytes)
   }
 
   implicit val cassandraConnector = connector
   implicit val readconf = readConf
-  private[this] lazy val baseRdd =
+
+  lazy val baseRdd =
     sqlContext.sparkContext.cassandraTable[CassandraSQLRow](tableRef.keyspace, tableRef.table)
 
   def buildScan(): RDD[Row] = baseRdd.asInstanceOf[RDD[Row]]
 
-  override def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row] = {
-    buildScan(requiredColumns, filters, Map.empty)
+  override def unhandledFilters(filters: Array[Filter]): Array[Filter] = filterPushdown match {
```

Comment:
No need for pattern matching. This is like `if(filterPushDown) .... else filters`